python/heterocl/tvm/runtime.py

```diff
@@ -20,6 +20,49 @@ def run_process(cmd, pattern=None, env=None):
     if debug: print("[DEBUG] Commands outputs: \n " + out.decode("utf-8"))
     return out.decode("utf-8")
 
+@register_func
+def exec_init(dev_hash, shmids, names):
+    # check whether pre-compiled bitstream exitsts
+    kernel = "project/kernel.cpp"
+    pre_compiled = False
+    if os.path.exists(kernel):
+        fp = open(kernel, "r")
+        if str(dev_hash) in fp.read():
+            pre_compiled = True
+            print("[{}] Skip codogen. Reuse pre-generated kernel code".format(
+                time.strftime("%H:%M:%S", time.gmtime())))
+        fp.close()
+
+    pre_compiled = False
+    # check whether compiled binary exists 
+    # re-compile if not. otherwise only compile host
+    if pre_compiled:
```

Comment:
You set `pre_compiled` to be `False` twice and it seems like it will only be `False`?

Reply:
This is fixed. Will push it later.